spec-0008/index.md

```diff
@@ -0,0 +1,227 @@
+This can be used by individual users and also in GitHub Actions workflows where the GitHub CLI utility is installed by default.
+
+```yaml
+- name: Verify sdist artifact attestation
+  env:
+    GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  run: gh attestation verify dist/<package name>-*.tar.gz --repo ${{ github.repository }}
+
+- name: Verify wheel artifact attestations
+  env:
+    GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  run: gh attestation verify dist/<package name>-*.whl --repo ${{ github.repository }}
+```
+
+### Adopt OIDC through the use of PyPI Trusted Publishers
+
+[Trusted Publishers](https://docs.pypi.org/trusted-publishers/) provide a way to securely establish a short lived authentication token between a project repository and a distribution platform &mdash; such as PyPI.
+It replaces the need to use a long lived token to authenticate, reducing the security risks associated with authentication tokens (e.g., tokens being compromised, the need to rotate tokens).
+
+Trusted Publishers can be used in GitHub Actions by using the [`pypa/gh-action-pypi-publish`](https://github.com/pypa/gh-action-pypi-publish) GitHub Action defaults in a GitHub Actions environment.
+
+```yaml
+jobs:
+  publish:
+    name: Publish release to PyPI
+    runs-on: ubuntu-latest
+    environment: publish-package
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    steps:
+      # retrieve your distributions here
+      # ...
+
+      - name: Publish distribution to PyPI
+        uses: pypa/gh-action-pypi-publish@<full action commit SHA> # vX.Y.Z
+        with:
+          print-hash: true
+```
+
+### Example workflow
+
+The following is a complete example of a workflow which can be used as a starting point:
+
+```yaml
+name: publish distributions
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  publish:
+    name: Publish Python distribution to PyPI
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      attestations: write
+    environment:
+      name: publish-package
+
+    steps:
+      # - name: Collect built artifacts
+      # ...
+
+      - name: Generate artifact attestation for sdist and wheels
+        uses: actions/attest-build-provenance@<full action commit SHA> # vX.Y.Z
+        with:
+          subject-path: "dist/<package name>-*"
+
+      - name: Verify sdist artifact attestation
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: gh attestation verify dist/<package name>-*.tar.gz --repo ${{ github.repository }}
+
+      - name: Verify wheel artifact attestations
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: gh attestation verify dist/<package name>-*.whl --repo ${{ github.repository }}
```

Comment:
```suggestion        
      - name: Verify artifact attestation
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        shell: bash
        run: |
          for artifact in dist/*; do
              echo "# ${artifact}"
              gh attestation verify "${artifact}" --repo ${{ github.repository }}
          done
```

This allows verification for all artifact in a loop (`xargs` might be more reasonable, but don't work with the current `gh attestation verify` implementation.